src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -46,6 +47,49 @@ def infer_schema_from_description(self, table_or_query: str) -> StructType:
             cursor.close()
             conn.close()
 
+    def udtf_ingestion(
+        self,
+        session: "snowflake.snowpark.Session",
+        schema: StructType,
+        partition_table: str,
+        external_access_integrations: str,
+        fetch_size: int = 1000,
+    ) -> "snowflake.snowpark.DataFrame":
+        session.udtf.register(
+            self.udtf_class_builder(fetch_size),
+            name="dbapi",
+            output_schema=StructType(
+                [
+                    StructField(field.name, StringType(), field.nullable)
+                    for field in schema.fields
+                ]
+            ),
+            external_access_integrations=[external_access_integrations],
+            packages=get_dbms_packages(self.create_connection()),
```

Comment:
should we expose a way to let users specify `packages` and `imports` by themselves? would this help custom data source scenario

Reply:
i think we can, but we need more discussion on how we expose it, I think we can do it in another PR